ponzu/llama/curatedTVL.py

```diff
@@ -0,0 +1,130 @@
+
+    def chain_protocols_tvb(self, chains, curated=False):
+        self._prepare_tvl(chains, curated)
+        tvb_protocols = {}
+        for chain in chains:
+            tvb_protocols_chain = self.tvl.loc[(self.tvl['type'] == 'borrowed') & (self.tvl['chain'] == chain)]
+            tvb_protocols[f'tvb_protocols_{chain}'] = pd.pivot_table(tvb_protocols_chain, values='tvl', index='date', columns='protocol', aggfunc=sum)
+        return tvb_protocols
+
+    def chain_liquid_staking(self, chains):
+        self._prepare_tvl(chains, False) 
+        df_liquid_staking = self.tvl[self.tvl['category'] == 'Liquid Staking']
+        return self._get_pivot_table('tvl', 'date', 'chain', sum, df_liquid_staking)
+    
+    def chain_protocols_liquid_staking(self, chains):
+        self._prepare_tvl(chains, False)
+        df_liquid_staking = self.tvl[self.tvl['category'] == 'Liquid Staking']
+        ls_protocols = {}
+        for chain in chains:
+            ls_chain = df_liquid_staking[df_liquid_staking['chain'] == chain]
+            ls_protocols[f'ls_protocols_{chain}'] = pd.pivot_table(ls_chain, values='tvl', index='date', columns='protocol', aggfunc=sum)
+        return ls_protocols
+    
+    def defi_diversity(self, chains, curated=False, threshold=0.9):
+        tvl_protocols = self.chain_protocols_tvl(chains, curated)
+        defi_diversity_chains = pd.DataFrame()
+        for chain in chains:
+            df = tvl_protocols[f'tvl_protocols_{chain}']
+            df = df.fillna(0)
+            df['total'] = df.sum(axis=1)
+            diversity = []
+            for index, row in df.iterrows():
+                row_values = row.values[:-1]
+                sorted_row = sorted(row_values, reverse=True)
+                cumsum = pd.Series(sorted_row).cumsum()
+                div_index = (cumsum <= threshold * row['total']).sum()
+                diversity_value = div_index + 1
+                diversity.append(diversity_value)
+
+            defi_diversity_protocol = pd.DataFrame({f'defi_diversity_{chain.lower()}': diversity}, index=df.index)
+
+            if defi_diversity_chains.empty:
+                defi_diversity_chains = defi_diversity_protocol
+            else:
+                defi_diversity_chains = pd.merge(defi_diversity_chains, defi_diversity_protocol, how='outer', on='date')
+
+        defi_diversity_chains.sort_index(ascending=True, inplace=True)
+
+        return defi_diversity_chains
+
+    
+    def chain_dex_volume(self, chain):
+        df = self.getFundamentalsByChains(chain, ['volume']) 
+        df = df[df['type'] == 'volume']
+        df_total = pd.pivot_table(df, values='value', index='date', columns='chain', aggfunc=sum)
+        df_total.index.name = 'DATE'
+        df_protocol = pd.pivot_table(df, values='value', index='date', columns='protocol', aggfunc=sum)
+        df_protocol.index.name = 'DATE'
+        return df_total, df_protocol
+
+    def llama_unique_chain(self):
```

Comment:
u lazy af -- why you need this

Reply:
I like just having to interact with one file instead of pulling from both the wrapper and base llama. but can delete if u think it's dumb

Reply:
but this inherits the base functions? 

just do peterLlama.getProtocols().chain.unique() in your downstream code

Reply:
didn't know it worked like this